src/freenas/usr/local/sbin/firmware_update_ses.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2020 iXsystems, Inc.
+# All rights reserved.
+# This file is a part of TrueNAS
+# and may not be copied and/or distributed
+# without the express permission of iXsystems.
+
+import json
+import logging
+import logging.config
+import logging.handlers
+import os
+import re
+import sys
+import subprocess
+
+UPDATE_FAILED_SENTINEL = "/data/.ses_firmware_update_failed"
+UPDATE_LOG = "/data/ses_firmware_update.log"
+UPDATE_PATH = "/usr/local/share/ses_firmware"
+
+logger = logging.getLogger(__name__)
+
+
+def run(args, check=True):
+    logger.debug("Running %r", args)
+    return subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, encoding="utf-8", errors="ignore",
+                          check=check)
+
+
+def write_update_failed_sentinel():
+    with open(UPDATE_FAILED_SENTINEL, "w"):
+        pass
+
+
+class Updater:
+    def __init__(self):
+        self.found = False
+
+        with open(os.path.join(UPDATE_PATH, "manifest.json")) as f:
+            manifest = json.load(f)
+
+        self.update_path = os.path.join(UPDATE_PATH, manifest["file"])
+        self.update_version = manifest["version"]
+
+        devlist = run(["camcontrol", "devlist"]).stdout
+        m = re.search(r"<CELESTIC R0904-F0001-01 ([0-9]+)>.*\((ses[0-9]+),", devlist)
```

Comment:
What expansion shelf is this? Is the a newer one? The ES60 shelves show up in camcontrol as `CELESTIC R0904-F1001`.

Also, do we want to alert that "Enclosure not found" for systems that don't have this enclosure?

Reply:
@darkfiberiru sorry, do you any news on this?